std/internal/cstring.d

```diff
@@ -123,6 +130,7 @@ auto tempCString(To = char, From)(From str)
 
     private:
         To* _ptr;
+        size_t _length;        // length of the string
```

Comment:
I had thought about that. But what could be simpler than using the [] operator to produce an array? It's the same thing we do with static arrays to convert them to dynamic ones.